### PR TITLE
fix: vulnerability in cover and admin uploads

### PR DESCRIPTION
**Pull request by barisusakli**

### CODE REVIEW

**Comment by ahmedsherif** [comment]
@barisusakli  Would you please explain more about the vulnerability and with possible demo ?

**Comment by julianlam** [comment]
@ahmedsherif For the safety and security of NodeBB instances available, we cannot (or rather, would _rather not_) release details until 30 days after the fix has been released.

File: src/image.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ image.stripEXIF = async function (path) {
 		const sharp = requireSharp();
 		await sharp(buffer, { failOnError: true }).rotate().toFile(path);
 	} catch (err) {
-		winston.error(err);
+		winston.error(err.stack);
```
Comment (julianlam) [inline src/image.js:104]: Intentional? I am not actually sure whether we can pass `err` to winston, I always just pass `err.message` but I vaguely recall that perhaps we updated it to allow handling the error object...

File: test/file.js [hunk 2]
```diff
@@ -96,6 +96,14 @@ describe('file', function () {
...
 			});
 		});
+
+		it('should error if folder is relative', function (done) {
+			file.saveFileToLocal(filename, '../../text', tempPath + '000000000', function (err) {
+				assert(err);
+				assert.strictEqual(err.message, '[[error:invalid-path]]');
+				done();
+			});
+		});
```
Comment (julianlam) [inline test/file.js:106]: Advise adding a test to ensure that `saveFileToLocal` also fails if absolute path does not start with `upload_path`

**Comment by julianlam** [comment]
Otherwise LGTM

File: src/image.js (hunk 1 above)
Comment (barisusakli) [reply src/image.js:104]: I can't remember the exact details but there are some issues with winston3.0 not logging error objects properly. More info https://github.com/winstonjs/winston/issues/1338#issuecomment-431053451

**Pull request merged**